common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -573,12 +573,10 @@ def source_library_values(self):
         """
         lib_tools = self.runtime.service(self, 'library_tools')
         user_perms = self.runtime.service(self, 'studio_user_permissions')
-        all_libraries = lib_tools.list_available_libraries()
-        if user_perms:
-            all_libraries = [
-                (key, name) for key, name in all_libraries
-                if user_perms.can_read(key) or self.source_library_id == unicode(key)
-            ]
+        all_libraries = [
```

Comment:
It was only `None` because of how this was set up before, which meant that sometimes the `studio_user_permissions` service was unavailable. In that case, we defaulted to ignoring permissions and letting the user access all libraries.

Now that has been fixed, so `studio_user_permissions` should always be available (in Studio). What's more, the behavior here has been fixed so that if the service is unavailable, it will cause an error rather than ignoring permissions and granting the user inappropriate access.